src/tests/e2e/pages/login.page.js

```diff
@@ -29,71 +29,95 @@ const sleep = milliseconds => {
+        const captchaCode = await anticaptcha.getResult(base64String)
+        if (captchaCode) {
+          log.info('Captcha solution: ' + captchaCode)
+          await googleLoginPopup.keyboard.type(captchaCode)
+          await googleLoginPopup.keyboard.press('Enter')
         } else {
-          return acc
+          log.error('Unable to find captcha solution')
         }
-      })
-      await googleLoginPopup.waitForFunction('document.querySelector("input")')
-      log.info('Google login popup email field loaded')
-      await sleep(1000)
+      }
+    }
 
-      await googleLoginPopup.keyboard.type(username)
-      await googleLoginPopup.keyboard.press('Enter')
+    await googleLoginPopup.waitForFunction('document.querySelector("input")')
+    log.info('Google login popup password field loaded')
 
-      while (true) {
-        // solve captcha
-        await sleep(1000)
+    await googleLoginPopup.keyboard.type(password)
+    await googleLoginPopup.keyboard.press('Enter')
+    log.info('password entered')
 
-        if (await googleLoginPopup.$('input[name=password]')) {
-          log.info('Password input field found, proceed to type password')
-          break
-        }
-        let captchaElementHandler = await googleLoginPopup.$('#captchaimg')
-        if (captchaElementHandler) {
-          console.log(captchaElementHandler)
-          log.info('Found captcha in login popup')
-          // base64String contains the captcha image's base64 encoded version
-          const base64String = await captchaElementHandler.screenshot({ encoding: 'base64' })
-          log.info('Solving captcha...')
-          const captchaCode = await anticaptcha.getResult(base64String)
-          if (captchaCode) {
-            log.info('Captcha solution: ' + captchaCode)
-            await googleLoginPopup.keyboard.type(captchaCode)
-            await googleLoginPopup.keyboard.press('Enter')
-          } else {
-            log.error('Unable to find captcha solution')
-          }
-        }
-      }
+    // default to treat every login as a new user
+    let newUser = true
+    try {
+      await sleep(3000)
+      // new user would get a authorization page
+      await expect(googleLoginPopup).toClick('span', { text: 'Allow', timeout: 3000 })
+    } catch (e) {
+      // this is not a new user if authorization screen is not displayed
+      newUser = false
+    }
 
-      await googleLoginPopup.waitForFunction('document.querySelector("input")')
-      log.info('Google login popup password field loaded')
-      await sleep(1000)
+    await sleep(2000)
+    await expect(page).toMatchElement('p', { text: 'Loading...', timeout: 100000 })
 
-      await googleLoginPopup.keyboard.type(password)
-      await googleLoginPopup.keyboard.press('Enter')
+    if (newUser) {
+      // new user would get a authorization page
+      await expect(page).toMatchElement('p', {
+        text: 'Setting up your account...',
+        timeout: 100000
```

Comment:
Reduce timeout to something reasonable rather than 100s

Reply:
fixed